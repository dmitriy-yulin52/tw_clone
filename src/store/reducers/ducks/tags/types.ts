
export enum LoadingState{
    LOADED = 'LOADED',
    LOADING = 'LOADING',
    ERROR = 'ERROR',
    NEVER = 'NEVER',

}

export interface Tag {
    id:string,
    fullName:string
    userName:string
    count:number
    text:string
}

export interface TagsState {
    items:Tag[]
    loadingState:LoadingState

}