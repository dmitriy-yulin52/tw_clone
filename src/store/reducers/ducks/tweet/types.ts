import {Tweet} from "../tweets/types";

export enum LoadingState{
    LOADED = 'LOADED',
    LOADING = 'LOADING',
    ERROR = 'ERROR',
    NEVER = 'NEVER',

}


export interface TweetDataState {
    data:Tweet | null
    loadingState:LoadingState

}