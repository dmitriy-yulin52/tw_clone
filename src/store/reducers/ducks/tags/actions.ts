import {Action} from 'redux'
import {LoadingState, Tag} from "./types";


export enum TagsActionsType {
    SET_ITEMS = 'tags/SET_ITEMS',
    FETCH_ITEMS = 'tags/FETCH_ITEMS',
    SET_LOADING_STATE = 'tags/SET_LOADING_STATE'
}

export type TagsActions = SetTagsType | SetTagsLoadingStateType | FetchTagsType

/** types-actions*/
export interface SetTagsType extends Action<TagsActionsType> {
    type: TagsActionsType.SET_ITEMS
    payload: Tag[]
}
export interface FetchTagsType extends Action<TagsActionsType> {
    type: TagsActionsType.FETCH_ITEMS
}
export interface SetTagsLoadingStateType extends Action<TagsActionsType> {
    type: TagsActionsType.SET_LOADING_STATE
    payload:LoadingState
}

/** actions */
export const setTweets = (payload: Tag[]): SetTagsType => {

    console.log(payload,'payload')
    return {
        type: TagsActionsType.SET_ITEMS,
        payload
    }
}
export const setTweetsLoadingState = (payload:LoadingState): SetTagsLoadingStateType => {
    return {
        type: TagsActionsType.SET_LOADING_STATE,
        payload
    }
}
export const fetchTweets = (): FetchTagsType => {
    return {
        type: TagsActionsType.FETCH_ITEMS,
    }
}
