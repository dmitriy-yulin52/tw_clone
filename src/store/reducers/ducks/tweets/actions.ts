import {Action} from 'redux'
import {LoadingState, TweetsState} from "./types";


export enum TweetsActionsType {
    SET_TWEETS = 'tweets/SET_TWEETS',
    FETCH_TWEETS = 'tweets/FETCH_TWEETS',
    SET_LOADING_STATE = 'tweets/SET_LOADING_STATE'
}

export type TweetsActions = SetTweetsType | SetTweetsLoadingStateType

/** types-actions*/
export interface SetTweetsType extends Action<TweetsActionsType> {
    type: TweetsActionsType.SET_TWEETS
    payload: TweetsState['items']
}
export interface FetchTweetsType extends Action<TweetsActionsType> {
    type: TweetsActionsType.FETCH_TWEETS
}
export interface SetTweetsLoadingStateType extends Action<TweetsActionsType> {
    type: TweetsActionsType.SET_LOADING_STATE
    payload:LoadingState
}

/** actions */
export const setTweets = (payload: TweetsState['items']): SetTweetsType => {
    return {
        type: TweetsActionsType.SET_TWEETS,
        payload
    }
}
export const setTweetsLoadingState = (payload:LoadingState): SetTweetsLoadingStateType => {
    return {
        type: TweetsActionsType.SET_LOADING_STATE,
        payload
    }
}
export const fetchTweets = (): FetchTweetsType => {
    return {
        type: TweetsActionsType.FETCH_TWEETS,
    }
}
