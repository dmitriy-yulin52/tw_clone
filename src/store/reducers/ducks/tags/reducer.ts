import {LoadingState, TagsState} from "./types";
import produce, {Draft} from "immer";
import {TagsActions, TagsActionsType} from "./actions";


const initialTweetsState: TagsState = {
    items: [],
    loadingState: LoadingState.NEVER
}


export const tagsReducer = produce((draft: Draft<TagsState>, action: TagsActions) => {


    switch (action.type) {
        case TagsActionsType.SET_LOADING_STATE:
            draft.loadingState = action.payload
            break
        case TagsActionsType.SET_ITEMS:
            draft.items = action.payload
            draft.loadingState = LoadingState.LOADED
            break
        case TagsActionsType.FETCH_ITEMS:
            draft.items = []
            draft.loadingState = LoadingState.LOADING
            break
        default:
            break
    }

}, initialTweetsState)