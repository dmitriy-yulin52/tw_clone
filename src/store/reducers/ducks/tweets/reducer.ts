import {LoadingState, TweetsState} from "./types";
import produce, {Draft} from "immer";
import {TweetsActions, TweetsActionsType} from "./actions";


const initialTweetsState: TweetsState = {
    items: [],
    loadingState: LoadingState.NEVER
}


export const tweetsReducer = produce((draft: Draft<TweetsState>, action: TweetsActions) => {


    switch (action.type) {
        case TweetsActionsType.SET_LOADING_STATE:
            draft.loadingState = action.payload
            break
        case TweetsActionsType.SET_TWEETS:
            draft.items = action.payload
            draft.loadingState = LoadingState.LOADED
            break
        case TweetsActionsType.FETCH_TWEETS:
            draft.items = []
            draft.loadingState = LoadingState.LOADING
            break
        default:
            break
    }

}, initialTweetsState)