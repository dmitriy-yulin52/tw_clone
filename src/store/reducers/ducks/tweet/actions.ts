import {Action} from 'redux'
import {LoadingState} from "./types";
import {Tweet} from "../tweets/types";


export enum TweetDataActionsType {
    SET_TWEET_DATA = 'tweet/SET_DATA',
    FETCH_TWEET_DATA = 'tweet/FETCH_DATA',
    SET_LOADING_STATE = 'tweet/SET_LOADING_STATE'
}

export type TweetDataActions = SetTweetDataType | SetTweetDataLoadingStateType | FetchTweetDataType

/** types-actions*/
export interface SetTweetDataType extends Action<TweetDataActionsType> {
    type: TweetDataActionsType.SET_TWEET_DATA
    payload: Tweet
}
export interface FetchTweetDataType extends Action<TweetDataActionsType> {
    type: TweetDataActionsType.FETCH_TWEET_DATA
    payload:string
}
export interface SetTweetDataLoadingStateType extends Action<TweetDataActionsType> {
    type: TweetDataActionsType.SET_LOADING_STATE
    payload:LoadingState
}

/** actions */
export const setTweetData = (payload: Tweet): SetTweetDataType => {
    console.log(payload,'setTweetData')
    return {
        type: TweetDataActionsType.SET_TWEET_DATA,
        payload
    }
}
export const setTweetDataLoadingState = (payload:LoadingState): SetTweetDataLoadingStateType => {
    return {
        type: TweetDataActionsType.SET_LOADING_STATE,
        payload
    }
}
export const fetchTweetData = (payload:string): FetchTweetDataType => {
    return {
        type: TweetDataActionsType.FETCH_TWEET_DATA,
        payload
    }
}
