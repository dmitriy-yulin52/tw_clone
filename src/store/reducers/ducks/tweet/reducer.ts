import {LoadingState, TweetDataState} from "./types";
import produce, {Draft} from "immer";
import {TweetDataActions, TweetDataActionsType} from "./actions";
import {Tweet} from "../tweets/types";


const initialTweetDataState: TweetDataState = {
    data: null,
    loadingState: LoadingState.NEVER
}


export const tweetReducer = produce((draft: Draft<TweetDataState>, action: TweetDataActions) => {


    switch (action.type) {
        case TweetDataActionsType.SET_LOADING_STATE:
            draft.loadingState = action.payload
            break
        case TweetDataActionsType.SET_TWEET_DATA:
            draft.data = action.payload
            draft.loadingState = LoadingState.LOADED
            break
        case TweetDataActionsType.FETCH_TWEET_DATA:
            // draft.data = []
            draft.loadingState = LoadingState.LOADING
            break
        default:
            break
    }

}, initialTweetDataState)