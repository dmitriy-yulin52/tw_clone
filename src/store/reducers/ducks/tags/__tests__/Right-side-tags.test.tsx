import {setTags} from "../actions";
import {tagsReducer} from "../reducer";
import {Tag, TagsState} from "../types";
import {LoadingState} from "../../tweets/types";


const tags: Tag[] = [
    {
        "id": "2",
        "fullName": "Sanya",
        "userName": "bolshoy",
        "text": "Non do nisi adipisicing ad ullamco nulla id adipisicing veniam occaecat pariatur aute adipisicing voluptate",
        "count": 2
    }
]


describe('tags test', () => {


       const initialTagsState: TagsState = {
            items: [],
            loadingState: LoadingState.NEVER
        }

    test('tagsReducer', () => {
        const data = {
            "payload": tags, "type": "tags/SET_ITEMS"
        }

        const action = setTags(tags)
        const end_state = tagsReducer(initialTagsState, action)

        expect(action).toEqual(data)
        expect(end_state.items[0].fullName).toBe('Sanya')
        expect(end_state.items.length).toEqual(1)
        expect(end_state).toMatchSnapshot()
    })
})