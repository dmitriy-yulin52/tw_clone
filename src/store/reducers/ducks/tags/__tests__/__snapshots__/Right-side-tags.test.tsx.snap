// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tags test tagsReducer 1`] = `
Object {
  "items": Array [
    Object {
      "count": 2,
      "fullName": "Sanya",
      "id": "2",
      "text": "Non do nisi adipisicing ad ullamco nulla id adipisicing veniam occaecat pariatur aute adipisicing voluptate",
      "userName": "bolshoy",
    },
  ],
  "loadingState": "LOADED",
}
`;
